python/ray/serve/schema.py

```diff
@@ -1042,3 +1043,18 @@ def _get_status(self) -> ServeStatus:
                 for app_name, app in self.applications.items()
             },
         )
+
+    def _serialize_bytes(self, values):
+        """Recursively convert bytes to base64 encoded str for JSON serialization."""
+        for key, value in values.items():
+            if isinstance(value, bytes):
+                values[key] = b64encode(value).decode("utf8")
+            elif isinstance(value, dict):
+                values[key] = self._serialize_bytes(value)
+        return values
+
+    def dict(self, *args, **kwargs):
```

Comment:
should dict func return dictionary?

Reply:
Yes, let me add a return type in the revision